src/main/recorder.ts

```diff
@@ -1,61 +1,48 @@
 import { Metadata } from './logutils';
 import { writeMetadataFile, runSizeMonitor, getNewestVideo, deleteVideo, cutVideo, addColor } from './util';
 import { mainWindow }  from './main';
-import { app } from 'electron';
 import path from 'path';
 import { AppStatus } from './types';
 
 const obsRecorder = require('./obsRecorder');
 const fs = require('fs');
 const glob = require('glob');
 
+type RecorderOptionsType = {
+    storageDir: string;
+    bufferStorageDir: string;
+    maxStorage: number;
+    monitorIndex: number;
+    audioInputDeviceId: string;
+    audioOutputDeviceId: string;
+    minEncounterDuration: number;
+};
+
 /**
  * Represents an OBS recorder object.
  */
  class Recorder {
     private _isRecording: boolean = false;
     private _isRecordingBuffer: boolean = false;
-    private _storageDir: string;
-    private _maxStorage: number;
-    private _bufferStorageDir: any;
     private _bufferRestartIntervalID?: any;
-    private _monitorIndex: number;
-    private _audioInputDeviceId: string;
-    private _audioOutputDeviceId: string;
-    private _minEncounterDuration: number;
-
+    private _options: RecorderOptionsType;
 
     /**
      * Constructs a new Recorder.
      */
-    constructor(
-        storageDir: string, 
-        maxStorage: number,
-        monitorIndex: number,
-        audioInputDeviceId: string,
-        audioOutputDeviceId: string,
-        minEncounterDuration: number
-    ) {
-        console.debug("[Recorder] Construcing recorder with: ", storageDir, maxStorage, monitorIndex);
-        this._storageDir = storageDir;
-        this._maxStorage = maxStorage;     
-        this._monitorIndex = monitorIndex;           
-        this._audioInputDeviceId = audioInputDeviceId;
-        this._audioOutputDeviceId = audioOutputDeviceId;
-        this._minEncounterDuration = minEncounterDuration;
-
-        // Something like: C:\Users\alexa\AppData\Local\Temp\WarcraftRecorder
-        this._bufferStorageDir = path.join(this._storageDir, ".temp"); 
-
-        if (!fs.existsSync(this._bufferStorageDir)) {
-            console.log("[Recorder] Creating dir:", this._bufferStorageDir);
-            fs.mkdirSync(this._bufferStorageDir);
+    constructor(options: RecorderOptionsType) {
+        this._options = options;
```

Comment:
Not sure how much I like this purely due to now all over this file we have this.something.somethingelse.

We could stick with having the member variables on this class as they were, and just split out options in the constructor to set them? I think that reads better throughout this file.

You could still set this._options for passing to the obsRecorder file.

Reply:
There's  lot of this.that.otherthing in many places and it personally doesn't bother me, but if you wish I can expand it to member variables.

Reply:
Fair enough - given the other comment happy to leave